src/main/java/org/openstreetmap/atlas/geography/MultiPolyLine.java

```diff
@@ -88,12 +88,34 @@ public Rectangle bounds()
         return Rectangle.forLocations(locations);
     }
 
+    @Override
+    public boolean equals(final Object other)
+    {
+        if (!(other instanceof MultiPolyLine))
+        {
+            return false;
+        }
+        final MultiPolyLine otherItem = (MultiPolyLine) other;
+        return this.hashCode() == otherItem.hashCode();
```

Comment:
pretty concerned about using this as a measure for equality. hash codes can potentially lead to both false negatives and false positives in certain cases

Reply:
I've changed hashCode to generate from concatenated string of polyLine's hashCode.

Reply:
@chunzc I think you can still run into integer overflow in the ```hashCode``` method below. Instead of relying on the hashCode you can probably compare the geometry. Maybe ```wkt``` if possible?

Reply:
ok, I will change it back to geometry comparison to apply equals on each polyline via iterator.